Bottom: 5b5e9db3d94df4c32490b5dc29c887400922bf3f
Top:    a2f94eb757d599b73edb2d5a9dc72456816a75fb
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-18 19:52:32 +0100

Refresh of give-the-hids-control-over-att

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 03fa2da..ea77412 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -58,9 +58,6 @@ ghid_port_ranges_changed (void)
 {
   GtkAdjustment *h_adj, *v_adj;
 
-  HideCrosshair ();
-  ghidgui->need_restore_crosshair = TRUE;
-
   h_adj = gtk_range_get_adjustment (GTK_RANGE (ghidgui->h_range));
   v_adj = gtk_range_get_adjustment (GTK_RANGE (ghidgui->v_range));
   gport->view_x0 = h_adj->value;
@@ -446,12 +443,10 @@ ghid_port_button_press_cb (GtkWidget * drawing_area,
   state = (GdkModifierType) (ev->state);
   mk = ghid_modifier_keys_state (&state);
   ghid_show_crosshair (FALSE);
-  notify_crosshair_change (false);
 
   do_mouse_action(ev->button, mk);
 
   ghid_invalidate_all ();
-  notify_crosshair_change (true);
   ghid_window_set_name_label (PCB->Name);
   ghid_set_status_line_label ();
   ghid_show_crosshair (TRUE);
@@ -472,13 +467,12 @@ ghid_port_button_release_cb (GtkWidget * drawing_area,
   state = (GdkModifierType) (ev->state);
   mk = ghid_modifier_keys_state (&state);
 
-  notify_crosshair_change (false);
-
   do_mouse_action(ev->button, mk + M_Release);
 
+  notify_crosshair_change (false);
   AdjustAttachedObjects ();
-  ghid_invalidate_all ();
   notify_crosshair_change (true);
+  ghid_invalidate_all ();
 
   ghid_window_set_name_label (PCB->Name);
   ghid_set_status_line_label ();
@@ -494,7 +488,6 @@ ghid_port_drawing_area_configure_event_cb (GtkWidget * widget,
 {
   static gboolean first_time_done;
 
-  HideCrosshair ();
   gport->width = ev->width;
   gport->height = ev->height;
 
@@ -528,7 +521,6 @@ ghid_port_drawing_area_configure_event_cb (GtkWidget * widget,
 
   ghid_port_ranges_scale (FALSE);
   ghid_invalidate_all ();
-  RestoreCrosshair ();
   return 0;
 }
 
diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index b2f0643..e21ef18 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -652,9 +652,8 @@ ghid_menu_cb (GtkAction * action, gpointer data)
     {
       notify_crosshair_change (false);
       AdjustAttachedObjects ();
-      ghid_invalidate_all ();
       notify_crosshair_change (true);
-      ghid_screen_update ();
+      ghid_invalidate_all ();
       ghid_window_set_name_label (PCB->Name);
       ghid_set_status_line_label ();
 #ifdef FIXME
